ravelin.js

```diff
@@ -895,172 +899,143 @@
-     */
-    RavelinJS.prototype.trackFingerprint = function(customerId) {
-      if (customerId) this.setCustomerId(customerId);
-      this._ravelin(['fingerprint'].concat(Array.prototype.slice.call(arguments, 1)));
-    }
-
-    /**
-     * setPublicAPIKey sets the API Key used to authenticate with Ravelin. It
-     * should be called before anything else.
-     *
-     * @param {String} apiKey
-     */
-    RavelinJS.prototype.setPublicAPIKey = function() {
-      this.apiKey = arguments[0];
-      this._ravelin(['setApiKey'].concat(Array.prototype.slice.call(arguments, 0)));
-    }
-
-    /**
-     * setCookieDomain configures where Ravelin will store any cookies on your
-     * domain. Set as high as possible, e.g. ".mysite.com" rather than
-     * ".www.uk.mysite.com".
-     *
-     * @param {String} domain
-     */
-    RavelinJS.prototype.setCookieDomain = function() {
-      this._ravelin(['setCookieDomain'].concat(Array.prototype.slice.call(arguments, 0)));
-    }
-
-    /**
-     * Set the customerId submitted with requests. This is used to associate device activity
-     * with a specific user.
-     *
-     * @param {String} customerId
-     */
-    RavelinJS.prototype.setCustomerId = function() {
-      this._ravelin(['setCustomerId'].concat(Array.prototype.slice.call(arguments, 0)));
-    }
-
-    /**
-     * Set the tempCustomerId submitted with requests. This is used as a temporary association between device/
-     * session data and a user, and should be followed with a v2/login request to Ravelin as soon as a
-     * customerId is available
-     *
-     * @param {String} customerId
-     */
-    RavelinJS.prototype.setTempCustomerId = function() {
-      this._ravelin(['setTempCustomerId'].concat(Array.prototype.slice.call(arguments, 0)));
-    }
-
-    /**
-     * Set the orderId submitted with requests. This is used to associate session-activity
-     * with a specific user.
-     *
-     * @param {String} orderId
-     */
-    RavelinJS.prototype.setOrderId = function() {
-      this._ravelin(['setOrderId'].concat(Array.prototype.slice.call(arguments, 0)));
-    }
-
-    RavelinJS.prototype._ravelin = function(args) {
-      if (typeof this.apiKey !== 'string') {
-        throw new Error("No tracking API key set. See RavelinJS.setPublicAPIKey");
-      }
-      if (!window.ravelin) {
-        // https://developer.ravelin.com/v2/#device-tracking.
-        (function(r,a,v,e,l,i,n){r[l]=r[l]||function(){(r[l].q=r[l].q||[]).push(arguments)};i=a.createElement(v);i.async=i.defer=1;i.src=e;a.body.appendChild(i)})(window, document, 'script', 'https://cdn.ravelin.net/js/rvn-beta.min.js', 'ravelin');
-      }
-
-      window.ravelin.apply(window, args);
+  RavelinJS.prototype._ravelin = function(args) {
+    if (typeof this.apiKey !== 'string') {
+      throw new Error('No tracking API key set. See RavelinJS.setPublicAPIKey');
+    }
+    if (!window) {
```

Comment:
Replaced the old 'if (window !== 'undefined') condition with just checking for the presence of window inside `_ravelin` calls. I wanted to remove that check so I could safely reorder these funcs, and figured the only genuine scenario this is falsey for is during testing, where we aren't testing fingerprint.js anyway.